scripts/volume-recovery

```diff
@@ -0,0 +1,147 @@
+#!/bin/bash
+
+set -e
+
+if [[ $# -ne 6 ]]; then
+  echo "Script needs 6 parameters, -s (snapshot id) -i (instance id) -v (volume id)"
+  exit 1
+fi
+
+while [[ $# -gt 0 ]]; do
+    key="${1}"
+    case ${key} in
+        -s|--snapshot)
+            SNAPSHOT="${2}"
+            shift
+            ;;
+        -i|--instance)
+            INSTANCE_ID="${2}"
+            shift
+            ;;
+        -v|--volume)
+            OLD_VOLUME="${2}"
+            shift
+            ;;
+        *)
+            log-output error "Unexpected option: ${1}"
+            usage
+            exit 1
+            ;;
+    esac
+    shift
+done
+
+mandatory=(
+  SNAPSHOT
+  INSTANCE_ID
+  OLD_VOLUME
+)
+for variable in ${mandatory[@]}; do
+  if [[ -z ${variable} ]]; then
+    echo "Mandatory variable missing: [${BOLD}${variable}${NORMAL}]"
+    exit 1
+  fi
+done
+
+# Check old volume and instance exist together
+VOLUME_CHECK_JSON=$(aws ec2 describe-volumes --volume-ids ${OLD_VOLUME})
+# Checking snapshot start time for validation
+SNAPSHOT_CHECK_JSON=$(aws ec2 describe-snapshots --snapshot-ids ${SNAPSHOT})
+# Bring back the instance name which is in the Name key tag
+INSTANCE_CHECK_JSON=$(aws ec2 describe-instances --instance-id ${INSTANCE_ID})
+
+INSTANCE_NAME=$(echo ${INSTANCE_CHECK_JSON} | jq -r '.Reservations[].Instances[].Tags[] | select(.Key == "Name").Value')
+INSTANCE_REGION_JSON=$(echo ${VOLUME_CHECK_JSON} | jq -r .Volumes[].AvailabilityZone)
+SNAPSHOT_START_TIME=$(echo ${SNAPSHOT_CHECK_JSON} | jq -r .Snapshots[].StartTime)
+VOLUME_INSTANCE_ID=$(echo ${VOLUME_CHECK_JSON} | jq -r .Volumes[].Attachments[].InstanceId)
+if [[ ${VOLUME_INSTANCE_ID} != ${INSTANCE_ID} ]]; then
+  INSTANCE_NAME=$(aws ec2 describe-instances --instance-id ${VOLUME_INSTANCE_ID} | jq -r '.Reservations[].Instances[].Tags[] | select(.Key == "Name").Value')
+  echo "Volume is not attached to this instance, it appears to be attached to ${INSTANCE_NAME} (${VOLUME_INSTANCE_ID})"
+  exit 1
+fi
+
+echo "Snapshot is ${SNAPSHOT}, which was created ${SNAPSHOT_START_TIME%.*}"
+echo "Instance is ${INSTANCE_ID}, the name is ${INSTANCE_NAME}"
+echo "Old volume is ${OLD_VOLUME}, which is in ${INSTANCE_REGION_JSON}"
```

Comment:
This line would need changing with the previous one (and any others that mention `${INSTANCE_REGION_JSON}`).